### PR TITLE
Safe persistance

### PR DESCRIPTION
**Pull request by jpalvarezl**
Closes https://github.com/gnosis/safe-android/issues/557 .

Changes proposed in this pull request:
- Added SafeDao and room implementation for the app
- Added basic use cases for the SafeRepository

@gnosis/mobile-devs

### CODE REVIEW

File: data/build.gradle [hunk 1]
```diff
@@ -1,6 +1,45 @@
 apply from: '../buildsystem/android_library.gradle'
 apply from: '../buildsystem/test_dependencies.gradle'
+apply from: '../buildsystem/test_android_dependencies.gradle'
+apply plugin: 'kotlin-kapt'
+
+android {
+    defaultConfig {
```
Comment (elgatovital) [inline data/build.gradle:7]: maybe :contracts is more suitable for these constants, as they relate to contracts

Comment (jpalvarezl) [reply data/build.gradle:7]: I tried but the `:contracts` module is a "java-library", I would have to change it to an android one. Are you ok with that?
Also, if we need to access this constants from :app, the :data module will always have to act as mediator between :app and :contracts. If you are fine with that I will push the change.

File: data/src/main/java/io/gnosis/data/repositories/SafeRepository.kt [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            }
+
+    private suspend fun getSafeBy(address: Solidity.Address): Safe? = safeDao.loadByAddress(address)
+
+    companion object {
+
+        @VisibleForTesting
+        const val ACTIVE_SAFE = "prefs.string.active_safe"
+
+        val safeMasterCopy_0_0_2 = BuildConfig.SAFE_MASTER_COPY_0_0_2.asEthereumAddress()!!
```
Comment (elgatovital) [inline data/src/main/java/io/gnosis/data/repositories/SafeRepository.kt:55]: constants should be upercase, what do you think?

File: data/src/test/java/io/gnosis/data/repositories/SafeRepositoryTest.kt [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+        preferences = spyk()
+        val application = mockk<Application>().apply {
+            every { getSharedPreferences(any(), any()) } returns preferences
+        }
+        val preferencesManager = PreferencesManager(application)
+        safeRepository = SafeRepository(safeDao, preferencesManager, ethereumRepository)
+    }
+
+    @Test
+    fun `getSafes - should return safes in the database`() = runBlocking {
```
Comment (elgatovital) [inline data/src/test/java/io/gnosis/data/repositories/SafeRepositoryTest.kt:37]: should we use test names without spaces here?

File: data/build.gradle (hunk 1 above)
Comment (elgatovital) [reply data/build.gradle:7]: I'm fine with the way it is right now. Having those constants directly in :app would also be good

File: data/src/main/java/io/gnosis/data/repositories/SafeRepository.kt (hunk 2 above)
Comment (jpalvarezl) [reply data/src/main/java/io/gnosis/data/repositories/SafeRepository.kt:55]: Sure, let's make them all SCREAMING_SNAKE_CASING?

File: data/src/test/java/io/gnosis/data/repositories/SafeRepositoryTest.kt (hunk 3 above)
Comment (jpalvarezl) [reply data/src/test/java/io/gnosis/data/repositories/SafeRepositoryTest.kt:37]: you mean like in instrumentation tests?

Comment (elgatovital) [reply data/src/test/java/io/gnosis/data/repositories/SafeRepositoryTest.kt:37]: I was thinking about having the same naming schema for all tests, but having spaces improves readability a lot. Let's use the same structure but spaces for unit tests and underscores for instrumentation.

**Pull request merged**